core/src/main/java/com/yahoo/imapnio/async/request/UidFetchCommand.java

```diff
@@ -36,7 +62,19 @@ public UidFetchCommand(@Nonnull final MessageNumberSet[] msgsets, @Nonnull final
      * @param items the data items
      */
     public UidFetchCommand(@Nonnull final String uids, @Nonnull final String items) {
-        super(true, uids, items);
+        this(uids, items, null);
+    }
+
+    /**
+     * Initializes a {@link UidFetchCommand} with string form uids and data items.
+     *
+     * @param uids the UID string following the RFC3501 syntax. For ex:3857529045,3857529047:3857529065
+     * @param items the data items
+     * @param partialExtUidFetchInfo partial extension uid fetch info
+     */
+    public UidFetchCommand(@Nonnull final String uids, @Nonnull final String items,
```

Comment:
not critical but a pattern. Let's not let user to pass in @Nullable partialExtUidFetchInfo, a nullable value makes parameter list looks confusing (required or not required? such). Change this to @Nonnull instead, and let line 65 to call super(...)